orangecontrib/text/vectorization/document_embedder.py

```diff
@@ -0,0 +1,185 @@
+"""This module contains classes used for embedding documents
+into a vector space.
+"""
+import zlib
+import base64
+import json
+import sys
+import warnings
+from typing import Tuple, Any, Optional
+import numpy as np
+
+from Orange.misc.server_embedder import ServerEmbedderCommunicator
+from orangecontrib.text import Corpus
+
+
+AGGREGATORS = ['mean', 'sum', 'max', 'min']
+LANGS_TO_ISO = {'English': 'en', 'Slovenian': 'sl', 'German': 'de'}
+LANGUAGES = list(LANGS_TO_ISO.values())
+
+class PretrainedEmbedder:
+    """This class is used for obtaining dense embeddings of documents in
+    corpus using fastText pretrained models from:
+    E. Grave, P. Bojanowski, P. Gupta, A. Joulin, T. Mikolov,
+    Learning Word Vectors for 157 Languages.
+    Proceedings of the International Conference on Language Resources and Evaluation, 2018.
+
+    Embedding is performed on server so the internet connection is a
+    prerequisite for using the class. Currently supported languages are:
+        - English (en)
+        - Slovenian (sl)
+        - German (de)
+
+    Attributes
+    ----------
+    language : str
+        ISO 639-1 (two-letter) code of desired language.
+    aggregator : str
+        Aggregator which creates document embedding (single
+        vector) from word embeddings (multiple vectors).
+        Allowed values are mean, sum, max, min.
+    """
+
+    def __init__(self, language: str = 'en',
+                 aggregator: str = 'mean') -> None:
+        lang_error = '{} is not a valid language. Allowed values: {}'
+        agg_error = '{} is not a valid aggregator. Allowed values: {}'
+        if language.lower() not in LANGUAGES:
+            raise ValueError(lang_error.format(language, ', '.join(LANGUAGES)))
+        self.language = language.lower()
+        if aggregator.lower() not in AGGREGATORS:
+            raise ValueError(agg_error.format(aggregator, ', '.join(AGGREGATORS)))
+        self.aggregator = aggregator.lower()
+
+        self._dim = None
+        self._embedder = _ServerEmbedder(self.aggregator,
+                                         model_name='fasttext-'+self.language,
+                                         max_parallel_requests=100,
+                                         server_url='https://example.com',
+                                         # TODO set proper url
+                                         embedder_type='text')
+
```

Comment:
I'd prefer the widget had a fixed size. You can squeeze here under __init__ by defining sizeHint: https://github.com/biolab/orange3/blob/d6cfa430ffb6860fcd447eddde6a50279529c131/Orange/widgets/data/owdatasets.py#L529